### PR TITLE
Derived order by

### PR DESCRIPTION
**Pull request by markstory**
Add a query cop that looks for `ORDER BY` inside queries that contain `COUNT(*)`. Today's incident was directly related to these kinds of queries causing really derpy query plans in MySQL.

### CODE REVIEW

File: tests/checks/test_order_by_count.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            )
+            stmt = sqlparse.parse(sql)[0]
+            assert False == self.has_order_by_count(stmt)
+
+            sql = (
+                "SELECT COUNT(*) FROM a LEFT JOIN b "
+                "USING (id)"
+            )
+            stmt = sqlparse.parse(sql)[0]
+            assert False == self.has_order_by_count(stmt)
```
Comment (alexBaizeau) [inline tests/checks/test_order_by_count.py]: what's the difference with the assertion on line 28?

**Comment by marvinpinto** [comment]
Looks reasonable to me :+1:

File: tests/checks/test_order_by_count.py (hunk 1 above)
Comment (kevinjqiu) [inline tests/checks/test_order_by_count.py]: Looks like copy pasta :)

**Comment by kevinjqiu** [comment]
LGTM :+1:

File: tests/checks/test_order_by_count.py (hunk 1 above)
Comment (markstory) [inline tests/checks/test_order_by_count.py]: It is copy pasta. I'll fix that up.

File: tests/checks/test_order_by_count.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                "SELECT COUNT(*) FROM a ORDER BY a.id DESC"
+            )
+            stmt = sqlparse.parse(sql)[0]
+            assert True == self.has_order_by_count(stmt)
+
+            sql = (
+                "SELECT COUNT(1) FROM a ORDER BY a.id DESC"
+            )
+            stmt = sqlparse.parse(sql)[0]
+            assert True == self.has_order_by_count(stmt)
```
Comment (alexBaizeau) [inline tests/checks/test_order_by_count.py]: That assertion looks like copy pasta too

Comment (markstory) [inline tests/checks/test_order_by_count.py]: One query is COUNT(*) The other is COUNT(1).

Comment (alexBaizeau) [inline tests/checks/test_order_by_count.py]: Ohhh :+1:

**Pull request merged**